cmd/load-generator/latency.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+)
+
+type point struct {
+	Sent     time.Time `json:"sent"`
+	Finished time.Time `json:"finished"`
+	Took     int64     `json:"took"`
+	PType    string    `json:"type"`
+	Action   string    `json:"action"`
+}
+
+type latencyFile struct {
+	metrics chan *point
+	f       *os.File
+	die     chan struct{}
+}
+
+func newLatencyFile(filename string) (*latencyFile, error) {
+	fmt.Printf("[+] Opening results file %s\n", filename)
+	file, err := os.OpenFile(filename, os.O_RDWR|os.O_APPEND|os.O_CREATE, os.ModePerm)
+	if err != nil {
+		return nil, err
+	}
+	f := &latencyFile{
+		metrics: make(chan *point, 2048),
+		die:     make(chan struct{}, 1),
+		f:       file,
+	}
+	go f.write()
+	return f, nil
+}
+
+func (f *latencyFile) write() {
+	for {
+		select {
+		case p := <-f.metrics:
+			data, err := json.Marshal(p)
+			if err != nil {
+				panic(err)
+			}
+			_, err = f.f.Write(append(data, []byte("\n")...))
```

Comment:
the `f.f` reads a little bit overly terse I think. Maybe the `latencyFile` member should be expanded to `outputFile` or something - thoughts?